docs/design/logrotate.md

```diff
@@ -0,0 +1,72 @@
+# Ceph CSI Log Rotate Design Document
+
+Log Rotate is the ability to rotate the logs by controlling the size of log files.
+With csi it will be includes to all the csi pods which includes csi daemons and provisoners.
+Using logrotate will create a log file at the hostpath and enable the logrotate sidecar container.
+Logrotate sidecar container will make sure that log files are rotated as per the inputs.
+
+So logrotate will be enabled by,
+
+`CephCSIOperatorConfig CRD`:
+```yaml
+spec:
+    provisioner:
+        privileged: true
+```
+
+Logroate configuration, 
+
+`CephCSIOperatorConfig CRD`:
+
+```yaml
+kind: CephCSIOperatorConfig 
+apiVersion: csi.ceph.io/v1alpha1
+….
+spec: 
+logLevel: 1 
+driverSpecDefaults: 
+logging: 
+    logLevel: 5
+    # one of: hourly, daily, weekly
+    Periodicity: daily
+    # SUFFIX may be 'M' or 'G' 
```

Comment:
why we need this comment?

Reply:
Just for user clarification, if it is to easy to understand I can remove it